test/integration/tcproute_test.go

```diff
@@ -432,6 +433,275 @@ func TestTCPRouteEssentials(t *testing.T) {
 	}, ingressWait, waitTick)
 }
 
+func TestTCPRouteReferencePolicy(t *testing.T) {
+	ns, cleanup := namespace(t)
+	defer cleanup()
+	t.Log("locking TCP port")
+	tcpMutex.Lock()
+	defer func() {
+		t.Log("unlocking TCP port")
+		tcpMutex.Unlock()
+	}()
+
+	other, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name()+"other")
```

Comment:
We've historically created namespaces with UUID's so that if a test were to be `kill -9`'d before cleanup, a subsequent run wouldn't collide with the previously killed one. Unless there's a specific reason, I think we should do the same here:

```suggestion
	other, err := clusters.GenerateNamespace(ctx, env.Cluster(), uuid.NewString())
```

Reply:
The helper provides that, the rest is just metadata for cleanup: https://github.com/Kong/kubernetes-testing-framework/blob/fa1c3009aa6273053049cf08cf236a8ec4df0e4d/pkg/clusters/utils.go#L145-L164

That said, there's no reason to clean these up separately--we can use the test name as the label for these also and let the regular post-test cleanup handle it.